### PR TITLE
[Enhancement] Remove optional Pool subscription from fast-unstake

### PR DESCRIPTION
**Pull request by ruseinov**
related: https://github.com/paritytech/substrate/issues/12337

### CODE REVIEW

File: frame/fast-unstake/src/lib.rs [hunk 1]
```diff
@@ -109,12 +106,7 @@ pub mod pallet {
 	pub struct Pallet<T>(_);
 
 	#[pallet::config]
-	pub trait Config:
-		frame_system::Config
-		+ pallet_staking::Config<
-			CurrencyBalance = <Self as pallet_nomination_pools::Config>::CurrencyBalance,
-		> + pallet_nomination_pools::Config
-	{
+	pub trait Config: frame_system::Config + pallet_staking::Config {
```
Comment (kianenigma) [inline frame/fast-unstake/src/lib.rs:109]: Later on, you can also tackle another part of #12337 and implement this via `trait StakingInterface` (if it supports all the needed APIs).

File: frame/fast-unstake/src/lib.rs [hunk 2]
```diff
@@ -392,48 +374,15 @@ pub mod pallet {
...
-				let ledger = match pallet_staking::Ledger::<T>::get(ctrl) {
-					Some(ledger) => ledger,
-					None => {
-						Self::deposit_event(Event::<T>::Errored { stash });
-						return <T as Config>::WeightInfo::on_idle_unstake()
-					},
-				};
-
-				let unstake_result = pallet_staking::Pallet::<T>::force_unstake(
+				let result = pallet_staking::Pallet::<T>::force_unstake(
```
Comment (kianenigma) [inline frame/fast-unstake/src/lib.rs:377]: much simpler now, nice.

**Review by kianenigma — APPROVED** [review]

**Review by rossbulat — APPROVED** [review]

File: frame/fast-unstake/src/lib.rs (hunk 1 above)
Comment (rossbulat) [inline frame/fast-unstake/src/lib.rs:109]: much cleaner

**Comment by ruseinov** [comment]
bit merge

**Comment by ruseinov** [comment]
bot merge

File: frame/fast-unstake/src/lib.rs (hunk 1 above)
Comment (ruseinov) [reply frame/fast-unstake/src/lib.rs:109]: will do

**Pull request merged**